mod-source-record-manager-server/src/main/java/org/folio/dao/JobExecutionDao.java

```diff
@@ -101,4 +103,14 @@ public interface JobExecutionDao {
    * @return collection of JobExecutionUserInfoCollection dtos with userIds, firstNames, lastNames
    */
   Future<JobExecutionUserInfoCollection> getRelatedUsersInfo(int offset, int limit, String tenantId);
+
+  /**
+   * Permanently deletes Job Executions from related tables depending upon difference in number of days
+   *
+   * @param tenantName tenant name
+   * @param diffNumberOfDays difference in Number of Days from the day record marked for deletion
+   * @return future with RowSet information
+   */
+
+  Future<RowSet<Row>> hardDeleteJobExecutions(String tenantName, long diffNumberOfDays);
```

Comment:
Question, why is a set of rows being returned? I would expect may be a `void` or boolean to signify that the hard delete was successful.

Reply:
I updated this method to return CompositeFuture, we returning future instead of void to be able to know when it is completed to write logs about results into periodical deletion job class. Also I've refactored approach to use transaction's wrapper from utility class.

Reply:
The final revision of this has been set to `Future<Boolean>`